### PR TITLE
Guarin lig 3049 add metric callback tests

### PR DESCRIPTION
**Pull request by guarin**
## Changes

* Add `MetricCallback` tests
* Refactor `MetricCallback` to keep track of metrics after train and validation epochs separately.

### CODE REVIEW

File: benchmarks/imagenet/resnet50/knn_eval.py [hunk 1]
```diff
@@ -56,15 +57,22 @@ def knn_eval(
...
         feature_dtype=torch.float16,
     )
+    metric_callback = MetricCallback()
     trainer = Trainer(
         max_epochs=1,
         accelerator=accelerator,
         devices=devices,
         logger=TensorBoardLogger(save_dir=str(log_dir), name="knn_eval"),
+        callbacks=[
+            DeviceStatsMonitor(),
```
Comment (guarin) [inline benchmarks/imagenet/resnet50/knn_eval.py:67]: Added `DeviceStatsMonitor` as we also use it for pretraining and linear evaluation.
